src/Tribe/Linked_Posts.php

```diff
@@ -760,54 +813,88 @@ public function handle_submission_by_post_type( $event_id, $linked_post_type, $s
+			 * @param string $linked_post_type Post type.
+			 * @param string $post_status      Post status.
+			 * @param int    $event_id         Post ID of the Event the Linked Post is attached to.
 			 */
 			$id = apply_filters( 'tribe_events_linked_post_create_' . $linked_post_type, null, $data, $linked_post_type, $post_status, $event_id );
 
 			/**
-			 * Filters the ID (default null) for creating posts from the event edit page
+			 * Filters the ID (default null) for creating posts from the event edit page.
 			 *
-			 * @param string $id Post type id index
-			 * @param array $data Data for submission
-			 * @param string $linked_post_type Post type
-			 * @param string $post_status Post status
-			 * @param int $event_id Post ID of the post the post type is attached to
+			 * @param string $id               Post type id index.
+			 * @param array  $data             Data for submission.
+			 * @param string $linked_post_type Post type.
+			 * @param string $post_status      Post status.
+			 * @param int    $event_id         Post ID of the Event the Linked Post is attached to.
 			 */
 			$id = apply_filters( 'tribe_events_linked_post_create', $id, $data, $linked_post_type, $post_status, $event_id );
 
-			if ( $id ) {
-				$linked_posts[] = $id;
+			if ( ! empty( $id ) ) {
+				$post_ids_to_link[] = $id;
 			}
 		}
 
-		// if we don't allow multiples, make sure there's only 1
-		if ( ! $this->allow_multiple( $linked_post_type ) && count( $linked_posts ) > 1 ) {
-			$linked_posts = array( $linked_posts[0] );
-		}
+		$post_ids_to_link = array_map( 'absint', $post_ids_to_link );
+		$post_ids_to_link = array_filter( $post_ids_to_link );
+		$post_ids_to_link = array_unique( $post_ids_to_link );
 
-		// if we allow multiples and there is more then one save current order
-		if ( $this->allow_multiple( $linked_post_type ) ) {
-			$this->order_linked_posts( $event_id, $linked_post_type, $submission[ $linked_post_type_id_field ] );
+		// If we do not allow multiples for this post type, ignore all but the first.
+		if (
+			! $this->allow_multiple( $linked_post_type )
+			&& 1 < count( $post_ids_to_link )
+		) {
+			$post_ids_to_link = array( $post_ids_to_link[0] );
 		}
 
-		$currently_linked_posts = $this->get_linked_posts_by_post_type( $event_id, $linked_post_type );
-		$currently_linked_posts = wp_list_pluck( $currently_linked_posts, 'ID' );
+		$prior_linked_posts = $this->get_linked_post_ids_by_post_type( $event_id, $linked_post_type );
 
-		$posts_to_add    = array_diff( $linked_posts, $currently_linked_posts );
-		$posts_to_remove = array_diff( $currently_linked_posts, $linked_posts );
+		$linked_post_type_meta_key = $this->get_meta_key( $linked_post_type );
 
-		foreach ( $posts_to_remove as $linked_post_id ) {
-			$this->unlink_post( $event_id, $linked_post_id );
-		}
+		if ( $prior_linked_posts === $post_ids_to_link ) {
+			// If the array values match type and value and are in the same order, no need to touch postmeta.
+			return;
+		} else {
```

Comment:
If this is the last part of this function you can avoid the else block  as you are returning already on the `if` block if executed.

Reply:
I reworked this multiple times and this is how it ended up. good catch!